.github/workflows/publish-provider-update.yml

```diff
@@ -1,17 +1,47 @@
 env:
   GITHUB_TOKEN: ${{ secrets.PULUMI_BOT_TOKEN }}
+  # The path within the Pulumi provider repo to the schema file.
   PROVIDER_SCHEMA_PATH: ${{ github.event.client_payload.schema-path }}
+  # The short-name of the provider, used in URL names. This is effectively the provider's
+  # identity in the Pulumi registry right now.
   PROVIDER_SHORT_NAME: ${{ github.event.client_payload.project-shortname }}
+  # The git tag that the provider is released with.
+  #
+  # This is only meaningful when the provider is based on a GitHub repo.
   PROVIDER_VERSION: ${{ github.event.client_payload.ref }}
+  # A public URL which hosts the provider's Pulumi schema.
+  #
+  # This URL must be durable, as it will be required available to build the registry.
+  PROVIDER_SCHEMA_URL: ${{ github.event.client_payload.schema-url }}
+  # A public URL which hosts the provider's  _index.md file.
+  #
+  # This URL must be durable, as it will be required available to build the registry.
+  PROVIDER_INDEX_URL:  ${{ github.event.client_payload.index-url }}
 
 name: provider docs build
 on:
   repository_dispatch:
     types:
+      # An update for a provider that is hosted in a Pulumi repo.
+      #
+      # Required inputs are:
+      # - PROVIDER_SHORT_NAME
+      # - PROVIDER_VERSION
+      #
+      # Optional inputs are:
+      # - PROVIDER_SCHEMA_PATH
       - resource-provider
+      # An update for a provider that is hosted opaquely.
```

Comment:
This is a new term - presumbably referring to third-party and dynamic providers? Could we make this clear?

Or just say "hosted elsewhere".

Reply:
I've changed the event name to `push-provider-update`.